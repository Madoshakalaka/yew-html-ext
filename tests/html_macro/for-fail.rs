use ::yew_html_ext::html;

mod smth {
    const KEY: u32 = 42;
}

fn main() {
    _ = html!{for x in};
    _ = html!{for x in 0 .. 10};
    _ = html!{for (x, y) in 0 .. 10 {
        <span>{x}</span>
    }};

    _ = html!{for _ in 0 .. 10 {
        <div key="duplicate" />
    }};

    _ = html!{for _ in 0 .. 10 {
        <div key={smth::KEY} />
    }};
}
