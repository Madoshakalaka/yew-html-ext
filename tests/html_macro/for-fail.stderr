error: unexpected end of input, expected an expression
 --> tests/html_macro/for-fail.rs:8:9
  |
8 |     _ = html!{for x in};
  |         ^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `html` (in Nightly builds, run with -Z macro-backtrace for more info)

error: unexpected end of input, expected curly braces
 --> tests/html_macro/for-fail.rs:9:9
  |
9 |     _ = html!{for x in 0 .. 10};
  |         ^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `html` (in Nightly builds, run with -Z macro-backtrace for more info)

error: duplicate key for a node in a `for`-loop
       this will create elements with duplicate keys if the loop iterates more than once
  --> tests/html_macro/for-fail.rs:15:18
   |
15 |         <div key="duplicate" />
   |                  ^^^^^^^^^^^

error: duplicate key for a node in a `for`-loop
       this will create elements with duplicate keys if the loop iterates more than once
  --> tests/html_macro/for-fail.rs:19:19
   |
19 |         <div key={smth::KEY} />
   |                   ^^^^

error[E0308]: mismatched types
  --> tests/html_macro/for-fail.rs:10:19
   |
10 |     _ = html!{for (x, y) in 0 .. 10 {
   |                   ^^^^^^    ------- this is an iterator with items of type `{integer}`
   |                   |
   |                   expected integer, found tuple
   |
   = note: expected type `{integer}`
             found tuple `(_, _)`
